reference/7.1/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md

```diff
@@ -122,6 +122,128 @@ Param(
+
+```powershell
+Param(
+    [Parameter(Mandatory=$false)]
+    [Switch]
+    $<ParameterName>
+)
+```
+
+Switch parameters are easy to use and are preferred over Boolean parameters,
+which have a more difficult syntax.
+
+For example, to use a switch parameter, the user types the parameter in the
+command.
+
+`-IncludeAll`
+
+To use a Boolean parameter, the user types the parameter and a Boolean value.
+
+`-IncludeAll:$true`
+
+When creating switch parameters, choose the parameter name carefully. Be sure
+that the parameter name communicates the effect of the parameter to the user.
+Avoid ambiguous terms, such as **Filter** or **Maximum** that might imply a
+value is required.
+
+## Dynamic parameters
+
+Dynamic parameters are parameters of a cmdlet, function, or script that are
+available only under certain conditions.
+
+For example, several provider cmdlets have parameters that are available only
+when the cmdlet is used in the provider drive, or in a particular path of the
+provider drive. For example, the **Encoding** parameter is available on the
+`Add-Content`, `Get-Content`, and `Set-Content` cmdlets only when it's used in
+a file system drive.
+
+You can also create a parameter that appears only when another parameter is
+used in the function command or when another parameter has a certain value.
+
+Dynamic parameters can be useful, but use them only when necessary, because
+they can be difficult for users to discover. To find a dynamic parameter, the
+user must be in the provider path, use the **ArgumentList** parameter of the
+`Get-Command` cmdlet, or use the **Path** parameter of `Get-Help`.
+
+To create a dynamic parameter for a function or script, use the `DynamicParam`
+keyword.
+
+The syntax is as follows:
+
+`DynamicParam {<statement-list>}`
+
+In the statement list, use an `If` statement to specify the conditions under
+which the parameter is available in the function.
+
+Use the `New-Object` cmdlet to create a
+**System.Management.Automation.RuntimeDefinedParameter** object to represent
+the parameter and specify its name.
+
+You can use a `New-Object` command to create a
+**System.Management.Automation.ParameterAttribute** object to represent
+attributes of the parameter, such as **Mandatory**, **Position**, or
+**ValueFromPipeline** or its parameter set.
+
+The following example shows a sample function with standard parameters named
+**Name** and **Path**, and an optional dynamic parameter named **DP1**. The
+**DP1** parameter is in the `PSet1` parameter set and has a type of `Int32`.
+The **DP1** parameter is available in the `Get-Sample` function only when the
+value of the **Path** parameter starts with `HKLM:`, indicating that it's being
+used in the `HKEY_LOCAL_MACHINE` registry drive.
+
+```powershell
+function Get-Sample {
+  [CmdletBinding()]
+  Param([String]$Name, [String]$Path)
+
+  DynamicParam
+  {
+    if ($Path.StartsWith("HKLM:"))
+    {
+      $attributes = New-Object -Type `
```

Comment:
I think you should promote use of:

```powershell
[System.Management.Automation.ParameterAttribute]::new()
```

instead of the cmdlet as it's slightly faster

Reply:
Done for all New-Object instances